packages/fx-core/src/plugins/resource/aad/utils/configs.ts

```diff
@@ -235,6 +400,113 @@ export class PostProvisionConfig {
   constructor(isLocalDebug = false) {
     this.isLocalDebug = isLocalDebug;
   }
+  public restoreConfigFromLocalSettings(localSettings: v2.LocalSettings): void {
+    const frontendEndpoint = localSettings.frontend?.tabEndpoint;
+    if (frontendEndpoint) {
+      this.frontendEndpoint = format(frontendEndpoint as string, Formats.Endpoint);
+    }
+    const botEndpoint = localSettings.bot?.botEndpoint;
+    if (botEndpoint) {
+      this.botEndpoint = format(botEndpoint as string, Formats.Endpoint);
+    }
+    const objectId = localSettings.auth?.objectId;
+    if (objectId) {
+      this.objectId = objectId as string;
+    }
+    if (objectId) {
+      this.objectId = objectId as string;
+    } else {
+      throw ResultFactory.SystemError(
+        GetConfigError.name,
+        GetConfigError.message(Errors.GetConfigError(ConfigKeys.objectId, Plugins.pluginName))
+      );
+    }
+    const applicationIdUri = localSettings.auth?.applicationIdUris;
+    if (applicationIdUri) {
+      this.applicationIdUri = applicationIdUri as string;
+    } else {
+      throw ResultFactory.SystemError(
+        GetConfigError.name,
+        GetConfigError.message(
+          Errors.GetConfigError(ConfigKeys.applicationIdUri, Plugins.pluginName)
+        )
+      );
+    }
+    const clientId = localSettings.auth?.clientId;
+    if (objectId) {
+      this.clientId = clientId as string;
+    } else {
+      throw ResultFactory.SystemError(
+        GetConfigError.name,
+        GetConfigError.message(Errors.GetConfigError(ConfigKeys.clientId, Plugins.pluginName))
+      );
+    }
+  }
+  public restoreConfigFromEnvInfo(ctx: v2.Context, envInfo: v3.EnvInfoV3): void {
+    const solutionSettings = ctx.projectSetting.solutionSettings as v3.TeamsFxSolutionSettings;
+    const aadResource = envInfo.state[BuiltInResourcePluginNames.aad] as v3.AADApp;
+    let frontendEndpoint = aadResource?.endpoint;
```

Comment:
Same as above. Will there be endpoint in aad state? Seems there is no code saving this.